onnxruntime/contrib_ops/cuda/bert/bert_padding.cu

```diff
@@ -361,6 +362,57 @@ void LaunchTrtSequenceOffset(int* trt_mha_padding_offset,
   }
 }
 
+__global__ void __launch_bounds__(kMAX_THREADS_PER_BLOCK)
+    getTrtSequenceOffset2d(int* trt_mha_padding_offset,
+                           const int* attention_masks,
+                           const int batch_size,
+                           const int sequence_length) {
+
+    typedef cub::BlockReduce<int, kMAX_THREADS_PER_BLOCK> BlockReduce;
+    __shared__ typename BlockReduce::TempStorage temp_storage;
+
+    const int batch_id = blockIdx.x;
+    const int* batch_mask = attention_masks + (batch_id * sequence_length);
+    int first_non_padding = sequence_length;
+    int last_non_padding = 0;
+    for (int i = threadIdx.x; i < sequence_length; i += blockDim.x) {
+      if (batch_mask[i])
+        if (first_non_padding > i) {
+          first_non_padding = i;
+        }
+        if (last_non_padding < i) {
+          last_non_padding = i;
+        }
+      }
+    }
+
+    int left_most = BlockReduce(temp_storage).Reduce(first_non_padding, cub::Min(), blockDim.x);
+    int right_most = BlockReduce(temp_storage).Reduce(last_non_padding, cub::Max(), blockDim.x);
+
+    if (threadIdx.x == 0) {
+      if (left_padding == 0) { // treat default as right padding
+        trt_mha_padding_offset[2 * batch_id] = (batch_id * sequence_length);
+        trt_mha_padding_offset[2 * batch_id + 1] = (batch_id * sequence_length) + right_most + 1;
+      } else {
+        trt_mha_padding_offset[2 * batch_id] = (batch_id * sequence_length) + left_most;
```

Comment:
trt_mha_padding_offset[2 * batch_id] shall always be batch_id * sequence_length no matter left side or right side padding.